### PR TITLE
[SSO] Use ClientConfigID & Subject as composite key for identities WEB-279

### PR DESCRIPTION
**Pull request by AlexTugarev**
During the Dedicated Onboarding flow, we learned that it's necessary to have the Issuer (URL) updatable, which means, it will be updatable on the Org Settings page as well, which mean it cannot be used as part of the stable composite key for the SSO identity in the data model. 

This PR updates the composite key of SSO identities to be `(ClientConfigID & Subject)`, which should not change during the lifecycle of a SSO config. (Transitioning from one SSO config to another is not in scope.)

## Related Issue(s)

Fixes WEB-279

## How to test
See unit test.

#### Preview status

gitpod:summary

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/server/src/iam/iam-session-app.ts [hunk 1]
```diff
@@ -81,17 +76,25 @@ export class IamSessionApp {
         };
     }
 
+    protected mapOIDCProfileToIdentity(payload: OIDCCreateSessionPayload): Identity {
```
Comment (easyCZ) [inline components/server/src/iam/iam-session-app.ts:79]: 👍

File: components/server/src/iam/iam-session-app.spec.ts [hunk 2]
```diff
@@ -165,6 +166,29 @@ class TestIamSessionApp {
             expect(result.body?.message).to.equal(c.expectedMessage);
         }
     }
+
+    @test public async testInvalidPayload_no_org_id() {
+        const payload = { ...this.payload };
+        console.log(JSON.stringify(payload));
```
Comment (easyCZ) [inline components/server/src/iam/iam-session-app.spec.ts:172]: drop log

File: components/public-api-server/pkg/oidc/service.go [hunk 3]
```diff
@@ -277,14 +277,16 @@ func (s *Service) Authenticate(ctx context.Context, params AuthenticateParams) (
 	}, nil
 }
 
-func (s *Service) CreateSession(ctx context.Context, flowResult *AuthFlowResult, organizationId string) (*http.Cookie, error) {
+func (s *Service) CreateSession(ctx context.Context, flowResult *AuthFlowResult, clientConfig *ClientConfig) (*http.Cookie, error) {
```
Comment (easyCZ) [inline components/public-api-server/pkg/oidc/service.go:280]: Did we not need to update any tests to make this pass?

**Review by easyCZ — APPROVED** [review]
LGTM, do we have a test for the session creation? If not, we should probs add one (can be follow-up)

File: components/public-api-server/pkg/oidc/service.go (hunk 3 above)
Comment (AlexTugarev) [reply components/public-api-server/pkg/oidc/service.go]: Added test for CreateSession. 
Next time we would.

File: components/server/src/iam/iam-session-app.spec.ts (hunk 2 above)
Comment (AlexTugarev) [reply components/server/src/iam/iam-session-app.spec.ts]: done

**Pull request merged**